pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -920,6 +944,63 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 				// If the container failed the liveness probe, we should kill it.
 				message = fmt.Sprintf("Container %s failed liveness probe", container.Name)
 			} else {
+				if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+					keepCount++
+					if container.Resources.Limits == nil || len(pod.Status.ContainerStatuses) == 0 {
+						continue
+					}
+					apiContainerStatus, exists := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name)
+					if !exists || apiContainerStatus.State.Running == nil ||
+						containerStatus.State != kubecontainer.ContainerStateRunning ||
+						containerStatus.ID.String() != apiContainerStatus.ContainerID ||
+						len(diff.ObjectDiff(apiContainerStatus.Resources.Limits, container.Resources.Limits)) == 0 {
+						continue
+					}
+					resizePolicy := make(map[v1.ResourceName]v1.ContainerResizePolicy)
+					for _, pol := range container.ResizePolicy {
+						resizePolicy[pol.ResourceName] = pol.Policy
+					}
+					restartContainer := false
+					determineContainerResize := func(rName v1.ResourceName, specValue, statusValue int64) bool {
+						if specValue == statusValue {
+							return false
+						}
+						if resizePolicy[rName] == v1.RestartContainer {
+							restartContainer = true
+							return true
+						}
+						cUpdateInfo := containerToUpdateInfo{
+							specIndex:           idx,
+							apiContainerStatus:  &apiContainerStatus,
+							kubeContainerStatus: containerStatus,
+							restart:             &restartContainer,
+						}
+						switch {
```

Comment:
nit: seems if statement would suffice

Reply:
yes. i tried it out after reading that for cases like this switch is faster.. maybe golang optimizes similar if-else-if.

Reply:
upstream review is fine with this.